plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/conditional/RelationalExpressionTests.java

```diff
@@ -0,0 +1,85 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.conditional;
+
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.processor.Processor;
+import org.elasticsearch.test.ESTestCase;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+
+public class RelationalExpressionTests extends ESTestCase {
+    IngestDocument ingestDocumentWithIntFoo = new IngestDocument("index", "type", "id", Collections.singletonMap("foo", 10));
+    IngestDocument ingestDocumentWithStringFoo = new IngestDocument("index", "type", "id", Collections.singletonMap("foo", "bar"));
+
+    public void testGte() {
+        RelationalExpression expression = new RelationalExpression("foo", "gte", 9);
+        expression.matches(ingestDocumentWithIntFoo);
+        assertTrue(expression.matches(ingestDocumentWithIntFoo));
+    }
+
+    public void testLte() {
+        RelationalExpression expression = new RelationalExpression("foo", "lte", 13);
+        expression.matches(ingestDocumentWithIntFoo);
+        assertTrue(expression.matches(ingestDocumentWithIntFoo));
+    }
+
+    public void testLt() {
+        RelationalExpression expression = new RelationalExpression("foo", "lt", 13);
+        expression.matches(ingestDocumentWithIntFoo);
+        assertTrue(expression.matches(ingestDocumentWithIntFoo));
+    }
+
+    public void testGt() {
+        RelationalExpression expression = new RelationalExpression("foo", "gt", 8);
+        expression.matches(ingestDocumentWithIntFoo);
+        assertTrue(expression.matches(ingestDocumentWithIntFoo));
+    }
+
+    public void testNumericComparisonWithoutNumberInDocument() {
+        RelationalExpression expression = new RelationalExpression("foo", "gt", 8);
+        assertFalse(expression.matches(ingestDocumentWithStringFoo));
+    }
+
+    public void testNumericComparisonWithoutNumberInOperand() {
+        RelationalExpression expression = new RelationalExpression("foo", "gt", "foo");
+        assertFalse(expression.matches(ingestDocumentWithIntFoo));
+    }
+
+    public void testEq() {
+        RelationalExpression expression = new RelationalExpression("foo", "eq", "bar");
+        expression.matches(ingestDocumentWithStringFoo);
+        assertTrue(expression.matches(ingestDocumentWithStringFoo));
+    }
+
+    public void testNeq() {
+        RelationalExpression expression = new RelationalExpression("foo", "neq", "hello");
+        expression.matches(ingestDocumentWithStringFoo);
+        assertTrue(expression.matches(ingestDocumentWithStringFoo));
+    }
```

Comment:
I think some randomisation would be great. We could pick random strings / numbers and verify is the result is the same  between the expression and the equals() or compareTo() methods on strings and numbers. For example:

```java
public void testRandomString() {
    String randomValue1 = randomUnicodeOfLength(6);
    String randomValue2 = randomUnicodeOfLength(6);
    IngestDocument doc = new IngestDocument("index", "type", "id", Collections.singletonMap("foo", randomValue1));
    RelationalExpression expression = new RelationalExpression("foo", "eq", randomValue2);
    assertThat(expression.matches(doc), equalTo(randomValue1.equals(randomValue2)));
}
```